compiler/rustc_infer/src/infer/combine.rs

```diff
@@ -852,13 +852,15 @@ pub trait ObligationEmittingRelation<'tcx>: TypeRelation<'tcx> {
     ///
     /// If they aren't equal then the relation doesn't hold.
     fn register_type_equate_obligation(&mut self, a: Ty<'tcx>, b: Ty<'tcx>) {
```

Comment:
```suggestion
    fn register_type_relate_obligation(&mut self, a: Ty<'tcx>, b: Ty<'tcx>) {
```

Reply:
and also probably a good idea to add a doc comment to this fn explaining where it gets the direction from

Reply:
I think it would be fine to describe how it's a property of the relation, and it should try to capture what relate behavior we would do with two rigid tys.